server/src/main/java/org/apache/druid/server/coordinator/duty/KillUnusedSegments.java

```diff
@@ -146,130 +156,128 @@ public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
-        }
+    for (String dataSource : dataSourcesToKill) {
+      if (submittedTasks >= availableKillTaskSlots) {
+        log.info(StringUtils.format(
+            "Submitted [%d] kill tasks and reached kill task slot limit [%d]. Will resume "
+            + "on the next coordinator cycle.", submittedTasks, availableKillTaskSlots));
+        break;
+      }
+      final DateTime maxUsedStatusLastUpdatedTime = DateTimes.nowUtc().minus(bufferPeriod);
+      final Interval intervalToKill = findIntervalForKill(dataSource, maxUsedStatusLastUpdatedTime, stats);
+      if (intervalToKill == null) {
+        datasourceToLastKillIntervalEnd.remove(dataSource);
+        continue;
+      }
 
-        try {
-          FutureUtils.getUnchecked(
-              overlordClient.runKillTask(
-                  TASK_ID_PREFIX,
-                  dataSource,
-                  intervalToKill,
-                  maxSegmentsToKill,
-                  maxUsedStatusLastUpdatedTime
-              ),
-              true
-          );
-          ++submittedTasks;
-          datasourceToLastKillIntervalEnd.put(dataSource, intervalToKill.getEnd());
-        }
-        catch (Exception ex) {
-          log.error(ex, "Failed to submit kill task for dataSource[%s] in interval[%s]", dataSource, intervalToKill);
-          if (Thread.currentThread().isInterrupted()) {
-            log.warn("Skipping kill task scheduling because thread is interrupted.");
-            break;
-          }
+      try {
+        FutureUtils.getUnchecked(
+            overlordClient.runKillTask(
+                TASK_ID_PREFIX,
+                dataSource,
+                intervalToKill,
+                maxSegmentsToKill,
+                maxUsedStatusLastUpdatedTime
+            ),
+            true
+        );
+        ++submittedTasks;
+        datasourceToLastKillIntervalEnd.put(dataSource, intervalToKill.getEnd());
+        remainingDatasourcesToKill.remove(dataSource);
+      }
+      catch (Exception ex) {
+        log.error(ex, "Failed to submit kill task for dataSource[%s] in interval[%s]", dataSource, intervalToKill);
+        if (Thread.currentThread().isInterrupted()) {
+          log.warn("Skipping kill task scheduling because thread is interrupted.");
+          break;
         }
       }
     }
 
-    if (log.isDebugEnabled()) {
-      log.debug(
-          "Submitted [%d] kill tasks for [%d] datasources.%s",
-          submittedTasks,
-          dataSourcesToKill.size(),
-          availableKillTaskSlots < dataSourcesToKill.size()
-              ? StringUtils.format(
-              " Datasources skipped: %s",
-              ImmutableList.copyOf(dataSourcesToKill).subList(submittedTasks, dataSourcesToKill.size())
-          )
-              : ""
-      );
-    }
+    log.info(
+        "Submitted [%d] kill tasks for [%d] datasources. Remaining datasources to kill: %s",
+        submittedTasks,
+        dataSourcesToKill.size(),
+        remainingDatasourcesToKill
```

Comment:
Super nit: I generally prefer this style as the logging occupies fewer lines without compromising readability, but either way is fine really.
```suggestion
        submittedTasks, dataSourcesToKill.size(), remainingDatasourcesToKill
```